chore(build):  Add a CI workflow to build and push multiubuntu image whenever a change detected

sikehish opened the pull request:
**Purpose of PR?**:
Add a CI workflow to build and push the multi-architecture Ubuntu-based image for KubeArmor on changes to the `main` branch.

Fixes #1876 

**Does this PR introduce a breaking change?**
No

**If the changes in this PR are manually verified, list down the scenarios covered:**:
- Verified that the CI workflow builds the Docker image successfully using multi-architecture (amd64 and arm64) support.
- Confirmed that the Docker image pushes correctly to Docker Hub when the workflow is triggered by `push` or `pull_request` on the `main` branch.


**Additional information for reviewer?** :
This PR introduces a new CI workflow and is not dependent on any previous PR or design.

**Checklist:**
- [ ] Bug fix. Fixes N/A
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [x] PR Title follows the convention of  `<type>(<scope>): <subject>`
- [ ] Commit has unit tests
- [ ] Commit has integration tests

<!--

The PR title message must follow convention:
`<type>(<scope>): <subject>`.

Where: <br />
- `type` is to define what type of PR is this.
  Most common types are:
    - `feat`      - for new features, not a new feature for build script
    - `fix`       - for bug fixes or improvements, not a fix for build script
    - `chore`     - changes not related to production code
    - `docs`      - changes related to documentation
    - `style`     - formatting, missing semi colons, linting fix etc; no significant production code changes
    - `test`      - adding missing tests, refactoring tests; no production code change
    - `refactor`  - refactoring production code, eg. renaming a variable or function name, there should not be any significant production code changes

- `scope` is a single word that best describes where the changes fit.
    - feature(`monitor`,`enforcer`)
    - test(`tests`, `bdd`)
    - chore(`build`)
- `subject` is a single line brief description of the changes made in the pull request.

-->

sikehish committed:
github/workflows: Created ci-build-and-push.yml

sikehish committed:
Revert "github/workflows: Created ci-build-and-push.yml"

This reverts commit 0129ca1e95fb0f6c3b428d2a2da3ba60c9ff75db.

sikehish committed:
github/workflows: Created ci-build-and-push.yml

sikehish committed:
github/workflows: Testing change in context in ci-build-and-push.yml

rksharma95 commented on .github/workflows/ci-build-and-push.yml:
```diff
@@ -0,0 +1,45 @@
+name: ci-build-and-push
```
```suggestion
name: ci-build-and-push-ubuntu
```

rksharma95 commented on .github/workflows/ci-build-and-push.yml:
```diff
@@ -0,0 +1,45 @@
+  push:
+    branches:
+      - main  
+  pull_request:
```
we don't want to push on pull request

sikehish replied:
I wasn't sure about it; thanks for the clarification!

rksharma95 commented on .github/workflows/ci-build-and-push.yml:
```diff
@@ -0,0 +1,45 @@
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
```
run on kubearmor repo only
```suggestion
if: github.repository == 'kubearmor/kubearmor'
```

rksharma95 commented on .github/workflows/ci-build-and-push.yml:
```diff
@@ -0,0 +1,45 @@
+          password: ${{ secrets.DOCKER_AUTHTOK }}
+
+      - name: Build and push multi-architecture image
+        uses: docker/build-push-action@v6
```
`examples/multiubuntu/build/build.sh` can handle build and push as well, no need to setup docker action for it.
build and push

daemon1024 replied:
We need multiarch builds, they are not handled in the script.

rksharma95 replied:
my bad, in that case keep repo name same that used in the script

sikehish replied:
> `examples/multiubuntu/build/build.sh` can handle build and push as well, no need to setup docker action for it. build and push

Do I need to make any changes? Do I need to use examples/multiubuntu/build/build.sh? A bit confused here. Would love for some clarity :))

rksharma95 replied:
@sikehish no need to use build.sh script, changes with your PR looks good just change image tag `kubearmor/ubuntu-w-utils:latest`

sikehish replied:
Hi @rksharma95 . Ive made the change. Let me know if you need me to change anything else :)

sikehish committed:
ci-build-and-push-ubuntu.yml: Renamed the workflow and added an if condition in build stage.

sikehish committed:
ci-build-and-push-ubuntu.yml: Renamed the workflow and added an if condition in build stage.

sikehish committed:
ci-build-and-push-ubuntu.yml: Changed image tag to kubearmor/ubuntu-w-utils:latest

sikehish committed:
chore(ci): add multiubuntu build path for e2e tests

Added examples/multiubuntu/build path in the CI workflow to run end-to-end tests for any changes.

sikehish committed:
Merge branch 'kubearmor:main' into main

sikehish committed:
ci-test-gingko.yml: Added functionality to build multiubuntu image locally for testing

Signed-off-by: sikehish <hisham0502@gmail.com>

sikehish committed:
Merge branch 'main' of https://github.com/sikehish/KubeArmor

Manik2708 commented on .github/workflows/ci-test-ginkgo.yml:
```diff
@@ -75,6 +77,11 @@ jobs:
+      - name: Build multiubuntu image
+        run: |
+          cd examples/multiubuntu/build
+          docker build -t kubearmor/multiubuntu:latest .
```
Please change the name to `kubearmor/ubuntu-w-utils:latest`

Manik2708 commented on .github/workflows/ci-test-ginkgo.yml:
```diff
@@ -84,6 +91,12 @@ jobs:
 
       - name: Run KubeArmor
         run: |
+          if [[ ${{ matrix.runtime }} == "containerd" ]]; then
```
Why another if statement? Is there any problem for putting in the same if statement?

sikehish replied:
> Why another if statement? Is there any problem for putting in the same if statement?

I thought it'd make it a bit more readable(as the kubearmor related images would be in a different block and the multiubuntu docker save block would be prominent enough. I'll make that change right away :)

sikehish committed:
ci-test-ginkgo.yml: Made minor refactoring related changes

Signed-off-by: sikehish <hisham0502@gmail.com>
